### DISCUSSION TITLE
Is there a Reduce layer?

### DISCUSSION THREAD

**Discussion by breuderink** [arogozhnikov/einops#126] [Q&A] [answered]
The API documentation states that there are three operations (rearrange, reduce, repeat), and two corresponding layers for different framework.

Now I would like to use a Repeat layer in PyTorch. What are my options?

**Comment by arogozhnikov** [comment, marked as answer]
did you mean "is there Repeat layer?". No, there is no such. In all cases I'm aware about broadcasting with Rearrange layer covers the need for Repeat. Can you describe a case when you need one?

**Reply by breuderink** [reply]
I indeed want to use a Repeat layer. Somehow I missed that I can do that with the Rearrange layer. I'll try that. Thanks.

**Reply by arogozhnikov** [reply]
Well, my point is:

```python
matrix = repeat(v, 'i -> i j', j=len(v)) + repeat(v, 'i -> j i', j=len(v)) 
```

is the same as 


```python
matrix = rearrange(v, 'i -> i 1') + rearrange(v, 'i -> 1 i') 
```
because of broadcasting, but the latter takes less space; 

in the context of layers I don't recall case when Repeat can't be replaced with Rearrange like in this example. So try using this
